cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/BrokerSetAwareGoal.java

```diff
@@ -161,6 +161,7 @@ protected void initGoalState(ClusterModel clusterModel, OptimizationOptions opti
     _brokerSetResolutionHelper = new BrokerSetResolutionHelper(clusterModel, brokerSetResolver);
     _brokersByBrokerSet = _brokerSetResolutionHelper.brokersByBrokerSetId();
     _replicaToBrokerSetMappingPolicy = _balancingConstraint.replicaToBrokerSetMappingPolicy();
+    _provisionResponse = new ProvisionResponse(ProvisionStatus.RIGHT_SIZED);
```

Comment:
Why do we set this to right sized? Wouldn't this cause the cluster never being reported as over provisioned if this goal is used?

Reply:
I guess you're right. I'll remove this line. However this does cause confusion to me, and maybe to following developers. Essentially you don't return RIGHT_SIZED even if your goal proves the cluster is neither OVER_PROVISIONED nor UNDER_PROVISIONED. I think it is confusing.

Reply:
For a goal to indicate whether a cluster is right sized, it must be able to gauge whether the cluster is over provisioned or not. Since the current implementation of this goal has no criteria to say whether the cluster is over provisioned or not, it cannot claim the cluster as right sized.